master/internal/resourcemanagers/resource_manager_config.go

```diff
@@ -0,0 +1,124 @@
+package resourcemanagers
```

Comment:
blocking: the split between this file and `resourcemanagers/config.go` is unclear. Unless there is a good reason for separating them they should be merged togerher.

Reply:
I don't really like merging all the configuration in one file because resource provider configurations are from the old world. Putting them in one file is very confusing to me. we should just entirely remove `config.go` in the future since we also putting `resource_pool_config.go` in the same directory.

Reply:
Imo this is pretty confusing, but if you really want to keep them separate you should at the least move everything that we are not planning to deprecate out of there (e.g, `MakeFitFunction` `MakeScheduler`)

Reply:
I agree with this. Right now the only reason the code layout makes sense is because of this comment. We should change the code (either rename stuff to indicate they are deprecated or add comments to the files)  so readers understand why they are separate.